src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -407,13 +407,11 @@ case class MatrixLiteral(
 }
 
 case class MatrixRead(
-  path: String,
-  spec: MatrixTableSpec,
+  typ: MatrixType,
+  override val partitionCounts: Option[Array[Long]],
   dropCols: Boolean,
-  dropRows: Boolean) extends MatrixIR {
-  def typ: MatrixType = spec.matrix_type
-
-  override def partitionCounts: Option[Array[Long]] = Some(spec.partitionCounts)
+  dropRows: Boolean,
+  f: (HailContext) => MatrixValue) extends MatrixIR {
 
```

Comment:
I think this f should be MatrixRead => MatrixValue

Reply:
I don't see why you'd want the input to be MatrixRead. Should I close this for now and we can discuss next week?

Reply:
sure, but basically we need to be able to push down dropRows here

Reply:
ok. that makes sense.